### PR TITLE
Mixpanel

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/278)

### CODE REVIEW

File: integrations/Mixpanel/browser.js [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+    delete props.distinct_id;
+    delete props.ip;
+    delete props.mp_name_tag;
+    delete props.mp_note;
+    delete props.token;
+
+    // Mixpanel People operations
+    if (this.people) {
+      // increment event count, check if the current event exists in eventIncrements
+      if (eventIncrements.indexOf(event) != -1) {
```
Comment (ruchiramoitra) [inline integrations/Mixpanel/browser.js:305]: !==

File: integrations/Mixpanel/browser.js [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+   * Since Mixpanel doesn't support lists of objects, invert each list of objects to a set of lists of object properties.
+   * Treats list transformation atomically, e.g. will only transform if EVERY item in list is an object
+   *
+   * @api private
+   * @param {Object} props
+   * @example
+   * input: {products: [{sku: 32, revenue: 99}, {sku:2, revenue: 103}]}
+   * output: {products_skus: [32, 2], products_revenues: [99, 103]}
+   */
+  inverseObjectArrays(input) {
```
Comment (ruchiramoitra) [inline integrations/Mixpanel/browser.js:418]: move these utility functions to another file? under this folder

File: integrations/Mixpanel/browser.js [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+    // eslint-disable-next-line camelcase
+    const user_id =
+      rudderElement.message.userId || rudderElement.message.anonymousId;
+    let { traits } = rudderElement.message.context;
+    const { email, username } = traits;
+    // id
+    if (user_id) window.mixpanel.identify(user_id);
+
+    // name tag
+    const nametag = email || username || id;
```
Comment (ruchiramoitra) [inline integrations/Mixpanel/browser.js:189]: what is `id` here?

File: integrations/Mixpanel/browser.js (hunk 2 above)
Comment (utsabc) [reply integrations/Mixpanel/browser.js:418]: Yep

File: integrations/Mixpanel/browser.js (hunk 3 above)
Comment (utsabc) [reply integrations/Mixpanel/browser.js:189]: Id is not required

**Review by ruchiramoitra — APPROVED** [review]

**Pull request merged**